app/controllers/application_controller.rb

```diff
@@ -2,8 +2,9 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  add_flash_types :success, :info, :warning, :danger
   helper_method :current_user, :time_format
-  before_action :get_categories, :set_cart
+  before_action :populate_nav, :set_cart, :authorize!
```

Comment:
this is kind of what I wanted to talk about - the use of before_action to take care of the authorization versus namespacing admin controllers...the functionality we have now seems like it's fine, but @karinamzalez is hesitant about using before_actions, and I wanted to see how you felt about their use. I think i remember you discouraging them in class, but I was hoping for a more in-depth answer, or maybe I just don't understand why they are bad.